src/librustc_mir_build/hair/pattern/check_match.rs

```diff
@@ -468,14 +468,14 @@ fn check_exhaustive<'p, 'tcx>(
     // In the case of an empty match, replace the '`_` not covered' diagnostic with something more
     // informative.
     let mut err;
+    let joined_patterns = joined_uncovered_patterns(&witnesses);
     if is_empty_match && !non_empty_enum {
         err = create_e0004(
             cx.tcx.sess,
             sp,
             format!("non-exhaustive patterns: type `{}` is non-empty", scrut_ty),
         );
     } else {
-        let joined_patterns = joined_uncovered_patterns(&witnesses);
```

Comment:
nit: afaict intentionally adds the usize/isize note even on the following case:
```rust
match 7usize {}
```
I personally think it's better if we only add these notes if there is at least one match arm

Reply:
It might also be a good idea to add a ui test for an empty integer match